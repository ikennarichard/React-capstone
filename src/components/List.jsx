import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styles from './List.module.css';

export default function List({
  co, no, no2, o3, so2, aqi, city, pm25,
}) {
  const [rating, setRating] = useState('');

  useEffect(() => {
    switch (aqi) {
      case 1: setRating('Good');
        break;
      case 2: setRating('Fair');
        break;
      case 3: setRating('Moderate');
        break;
      case 4: setRating('Poor');
        break;
      case 5: setRating('Very Poor');
        break;
      default: setRating('');
    }
  }, [aqi]);

  return (
    <li>
      <div>
        <h2 className={styles.subHeading}>
          { `${city} State` }
          <br />
          <small>{`Air Quality Index: ${aqi} (${rating})`}</small>
        </h2>
        <p className={styles.listItem}>
          Particulates PM2.5
          <span>{` ${pm25} μg/m3`}</span>
        </p>

        <p>
          Carbon:
          <span>{`${co} μg/m3`}</span>
        </p>
        <p>
          Nitrogen Dioxide:
          <span>{`${no2} μg/m3`}</span>
        </p>
        <p>
          Nitrogen Monoxide:
          <span>{`${no} μg/m3`}</span>
        </p>
        <p>
          Sulpher Dioxide:
          <span>{`${so2} μg/m3`}</span>
        </p>
        <p>
          Ozone:
          <span>{`${o3} μg/m3`}</span>
        </p>
      </div>
    </li>
  );
}

List.propTypes = {
  co: PropTypes.number.isRequired,
  no: PropTypes.number.isRequired,
  no2: PropTypes.number.isRequired,
  o3: PropTypes.number.isRequired,
  so2: PropTypes.number.isRequired,
  aqi: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  pm25: PropTypes.number.isRequired,
};
