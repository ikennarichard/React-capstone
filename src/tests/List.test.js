import renderer from 'react-test-renderer';
import List from '../components/List';

it('renders list item correctly', () => {
  const tree = renderer.create(
    <List co={3} no={3} no2={3} o3={4} so2={5} aqi={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
