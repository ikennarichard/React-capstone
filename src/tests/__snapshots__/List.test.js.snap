// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list item correctly 1`] = `
<li>
  <div>
    <h2
      className="subHeading"
    >
      undefined State
      <br />
      <small>
        Air Quality Index: 3 ()
      </small>
    </h2>
    <p
      className="listItem"
    >
      Particulates PM2.5
      <span>
         undefined μg/m3
      </span>
    </p>
    <p>
      Carbon:
      <span>
        3 μg/m3
      </span>
    </p>
    <p>
      Nitrogen Dioxide:
      <span>
        3 μg/m3
      </span>
    </p>
    <p>
      Nitrogen Monoxide:
      <span>
        3 μg/m3
      </span>
    </p>
    <p>
      Sulpher Dioxide:
      <span>
        5 μg/m3
      </span>
    </p>
    <p>
      Ozone:
      <span>
        4 μg/m3
      </span>
    </p>
  </div>
</li>
`;
